src/gas_price.rs

```diff
@@ -69,6 +69,26 @@ impl GasPrice1559 {
     }
 }
 
+impl std::fmt::Display for GasPrice1559 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let format_unit = |wei| {
+            let gwei: f64 = wei / 1e9;
+            if gwei >= 1.0 {
+                format!("{:.2} Gwei", gwei)
+            } else {
+                format!("{wei} wei")
+            }
+        };
+        write!(
+            f,
+            "{{ max_fee: {}, max_priority_fee: {}, base_fee: {} }}",
```

Comment:
One thing I hadn't considered is that this will not print `GasPrice` so you would always have to add that in logs that want to print that.
Are we fine with that?

Reply:
Fine with me. 👌